infrastructure/zk/src/config.ts

```diff
@@ -180,9 +181,10 @@ export function pushConfig(environment?: string, diff?: string) {
     env.modify('DATABASE_MERKLE_TREE_BACKUP_PATH', `./db/${environment}/backups`, l2InitFile, false);
 
     if (process.env.CONTRACTS_DEV_PROTOCOL_VERSION) {
+        const minor = unpackStringSemVer(process.env.CONTRACTS_DEV_PROTOCOL_VERSION)[1];
         env.modify(
             'CONTRACTS_GENESIS_PROTOCOL_VERSION',
-            (parseInt(process.env.CONTRACTS_GENESIS_PROTOCOL_VERSION!) + 1).toString(),
+            `0.${minor + 1}.0`, // The major version is always 0 for now
```

Comment:
It is better to parse the version, since after some upgrades are done the "true" genesis version may become a version with some patch number.